### ISSUE TITLE
Jacobian generation in ScalableTestGrids tests does not generate the sparsity pattern

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#13058]
Follow-up of #12937.

Jacobian generation in ScalableTestGrids tests does not generate the sparsity pattern. As a consequence, Type1_N_2_M_2, which has 10,000 equations, takes 0.16 to generate the Jacobian, while Type1_N_4_M_4, which has 50,000 equations, takes 4.5 s, with an O(N^2) scaling, which is inevitable if you generate dense Jacobian. 

This should be fixed if we want to reach and exceed the 1M equation threshold.

**Comment by phannebohm** [comment]
I think this is an appropriate MWE
```modelica
model MWE
  Real x,y;
equation
  der(x) * der(y) = x;
  der(x) + der(y) = y;
end MWE;
```
When generating the sparsity pattern, we lookup dependencies in a map, but the map contains wrong elements so it fails. The elements have wrong prefixes:
```
idx_map: {($SEED_ODE_JAC.y, 0), ($SEED_ODE_JAC.x, 1)}
dependencies: {$DER.y, $DER.x}
```
I suppose, at some point we populate the dependencies wrong and they should indeed be seeds, but my interpretation might be wrong. I will discuss this with @kabdelhak.

**Comment by phannebohm** [comment]
We might have fixed it with #13340. Let's see what the regressions say.

**Comment by casella** [comment]
Started job [#8762](https://test.openmodelica.org/jenkins/job/LibraryTesting/job/LibraryTest/8762/console)

**Comment by casella** [comment]
Unfortunately it didn't help much, see [regression report](https://libraries.openmodelica.org/branches/history/newInst-newBackend/2024-12-04%2015:46:06..2024-12-05%2015:28:29.html). In fact, the ScalableTestGrids models that were compiling now fail at the SimCode stage.

**Comment by casella** [comment]
Maybe you should try out some of those models before making the PR 😃

**Comment by casella** [comment]
@phannebohm reports that the apparent regression is actually an improvement, because previously sparsity pattern generation was not even attempted, now it is mostly working, save for some bugs that will be fixed soon.

**Comment by casella** [comment]
Update: after #13340, the [MWE](https://github.com/OpenModelica/OpenModelica/issues/13058#issuecomment-2515080951) by @phannebohm runs successfully with the NB.

Unfortunately, ScalableTestGrids models cannot yet be compiled. If you check, e.g., [ScalableTestGrids.Models.Type1.Type1_N_1_M_1](https://libraries.openmodelica.org/branches/newInst-newBackend/ScalableTestGrids_noopt/files/ScalableTestGrids_noopt_ScalableTestGrids.Models.Type1.Type1_N_1_M_1.err), the following error is reported during the Templates phase:
```
[OpenModelica/OMCompiler/Compiler/SimCode/SerializeModelInfo.mo:116:9-116:84:writable]
Error: Internal error SerializeModelInfo.serialize failed
[CodegenCFunctions.tpl:2998:11-2998:11:writable]
Error: Template error: Left hand side of an assignment is a record expression. Complex(0.0, 0.0).
```
@phannebohm is this still related to sparsity pattern generation, or is it another issue?

Thanks!

**Comment by casella** [comment]
I guess the issue regarding the sparsity pattern can be considered fixed. I'll open a separate one regarding the LHS Complex record. See #13980.
